tests/ui/drop/lint-tail-expr-drop-order.rs

```diff
@@ -0,0 +1,33 @@
+//@compile-flags: -Z unstable-options
+//@edition:2024
+
+#![deny(tail_expr_drop_order)]
+#![feature(shorter_tail_lifetimes)]
+
+struct LoudDropper;
+impl Drop for LoudDropper {
+    fn drop(&mut self) {
+        println!("loud drop")
+    }
+}
+impl LoudDropper {
+    fn get(&self) -> i32 {
+        0
+    }
+}
+
+fn should_lint() -> i32 {
+    let x = LoudDropper;
+    // Should lint
+    x.get() + LoudDropper.get()
+    //~^ ERROR: these values and local bindings have significant drop implementation that will have a different drop order from that of Edition 2021
+    //~| WARN: this changes meaning in Rust 2024
+}
+
+fn should_not_lint() -> i32 {
+    let x = LoudDropper;
+    // Should not lint
+    x.get()
+}
+
```

Comment:
Question: could we also add test cases for more complex examples?

e.g. should these cases warn / not warn?

```rs
fn more_cases() {
    fn inner() {
        let y = LoudDropper;
        y.get() + LoudDropper.get()
    }

    let closure = || { let y = LoudDropper; y.get() + LoudDropper.get() };

    match (...) {
        Ok(_) => {},
        Err() => { let y = LoudDropper; y.get() + LoudDropper.get() }
    }

    if ({let y = LoudDropper; y.get() + LoudDropper.get()} == 0) {}
}
```

Reply:
I have added the test for the nested items, closures, match and block in the tail expression location.